src/TeleportJoin.sol

```diff
@@ -259,12 +266,24 @@ contract TeleportJoin {
         require(batchedDaiToFlush <= uint256(type(int256).max), "TeleportJoin/overflow");
         daiJoin.join(address(this), batchedDaiToFlush);
         if (vat.live() == 1) {
-            (, uint256 art_) = vat.urns(ilk, address(this)); // rate == RAY => normalized debt == actual debt
-            uint256 amtToPayBack = _min(batchedDaiToFlush, art_);
+            (uint256 ink_, uint256 art_) = vat.urns(ilk, address(this)); // rate == RAY => normalized debt == actual debt
+            
+            // Fix any permissionless repays that may have occurred
+            if (art_ < ink_) {
+                address vow_ = vow;
+                uint256 diff;
+                unchecked {
+                    diff = ink_ - art_;
+                }
+                vat.frob(ilk, address(this), address(this), vow_, 0, _int256(diff)); // After this call the urn's art will be equal to its ink
+            }
+
+            int256 debt_ = debt[sourceDomain];
+            uint256 amtToPayBack = _min(batchedDaiToFlush, debt_ > 0 ? uint256(debt_) : 0);
             vat.frob(ilk, address(this), address(this), address(this), -int256(amtToPayBack), -int256(amtToPayBack));
             vat.slip(ilk, address(this), -int256(amtToPayBack));
```

Comment:
This could be behind an if conditional of `amtToPayBack > 0` to save a fair bit of gas. My guess though is the negative debt case will be unlikely, so it's probably fine to leave as-is.